### DISCUSSION TITLE
using custom embedding model

### DISCUSSION THREAD

**Discussion by hbredewold** [MaartenGr/BERTopic#1934] [Q&A] [answered]
Hi all!

Very new to topic modeling! For my master´s thesis, I´m using BERTopic to topic model tweets. Also, absolutely new to models, NLP, even python. Since the tweets are in Dutch, I would like to use a transformer model that is dutch, namely; https://huggingface.co/GroNLP/bert-base-dutch-cased. 

The getting started page for BERTopic tells me to do it like this:

```
from bertopic import BERTopic
topic_model = BERTopic(embedding_model="all-MiniLM-L6-v2")
```

However, the page for the specific model i want to use tells me to do this:

```
from transformers import AutoTokenizer, AutoModel, TFAutoModel

tokenizer = AutoTokenizer.from_pretrained("GroNLP/bert-base-dutch-cased")
model = AutoModel.from_pretrained("GroNLP/bert-base-dutch-cased")  # PyTorch
model = TFAutoModel.from_pretrained("GroNLP/bert-base-dutch-cased")  # Tensorflow
```

Is there anyone who has any idea to make this work?

Thanks in advance!

**Comment by MaartenGr** [comment, marked as answer]
I generally would not advise using a Transformer model that was not specifically trained to generate embeddings. Instead, I would advise looking at the [MTEB leaderboard](https://huggingface.co/spaces/mteb/leaderboard) and choosing a **multilingual** model that scores relatively high on the clustering metric. You can generally use that model in sentence-transformers.
